GVFS/GVFS.UnitTests/Common/PathsTests.cs

```diff
@@ -31,5 +40,44 @@ public void CanConvertOSPathToGitFormat()
             string doubleTransformedPath = Paths.ConvertPathToGitFormat(actualTransformedPath);
             doubleTransformedPath.ShouldEqual(expectedGitPath);
         }
+
+        [TestCase]
+        public void GetFilesRecursive()
+        {
+            string rootDirectory = Path.Combine(Path.GetTempPath(), Guid.NewGuid().ToString("N"));
+
+            List<string> expectedFiles = new List<string>
+            {
+                "file1.txt",
+                "file2.txt",
+                "a/file1.txt",
+                "a/file2.txt",
+                "a/b/file1.txt",
+                "a/b/file2.txt",
+                "a/b/c/file1.txt",
+                "a/b/c/file2.txt",
+                "a/b/c/d/file1.txt",
+                "a/b/c/d/file2.txt"
+            };
+
+            CreateEmptyFiles(rootDirectory, expectedFiles);
+
+            List<string> actualFiles = Paths.GetFilesRecursive(rootDirectory).ToList();
+
+            CollectionAssert.AreEquivalent(expectedFiles, actualFiles);
+        }
+
+        private static void CreateEmptyFiles(string rootDirectory, IEnumerable<string> files)
+        {
+            foreach (string file in files)
+            {
+                string fullFilePath = Path.Combine(rootDirectory, file);
+
+                string parentDirectory = Path.GetDirectoryName(fullFilePath);
+                Directory.CreateDirectory(parentDirectory);

```

Comment:
Unit tests must not do actual IO. Either mock out the file system, or drop this test and rely on functional tests.

Reply:
See my other note, I would make the new relative path method private anyway. Functional test coverage should be adequate here.

Reply:
I'll drop it and rely on functional tests (given it's being pushed into the repair job as private).